sdk/logical/storage_view.go

```diff
@@ -9,37 +9,74 @@ import (
 	"strings"
 )
 
-type StorageView struct {
-	storage Storage
-	prefix  string
+type StorageViewCore interface {
+	Prefix() string
+	SubView(prefix string) StorageView
+	SanityCheck(key string) error
+	ExpandKey(suffix string) string
+	TruncateKey(full string) string
+}
+
+type StorageView interface {
+	Storage
+	StorageViewCore
+}
+
+type TransactionalStorageView interface {
+	TransactionalStorage
+	StorageViewCore
+}
+
+// Note that, within a transaction, creating and committing from a
+// SubView commits the entire transaction.
+type StorageViewTransaction interface {
+	Transaction
+	StorageViewCore
 }
 
 var ErrRelativePath = errors.New("relative paths not supported")
 
-func NewStorageView(storage Storage, prefix string) *StorageView {
-	return &StorageView{
+func NewStorageView(storage Storage, prefix string) StorageView {
```

Comment:
Since this is called from `SubView`, we need to make sure it also handles `Transaction`, creating a `StorageViewTransaction` instead.

Reply:
`SubView` more generally is also one of the few storage layers that we'd expect to occur within a transaction. Things like the encrypted storage view (used by K/V plugin) are less likely to occur (as you'd probably wrap the entire outer storage entry). Thus, exposing `Commit`/`Rollback` on the `SubView` directly might be nicer for callers.